posit-dev/positron
# Get environment variables from R process for tasks

Addresses https://github.com/posit-dev/positron/issues/2723

Together with https://github.com/posit-dev/ark/pull/507, this sets up infrastructure so we can propagate known sets of environment variables to our tasks that use `ShellExecution` (or `ProcessExecution`). I added the `TESTTHAT` environment variables for `devtools::test()` only, for now. We can add other sets of environment variables as they come up.

### TODO:

- [x] Bump ark

### QA Notes

To QA this, you'll need an R package with some failing tests. [Here's one!](https://github.com/juliasilge/vamplyr)

- Run `testthat::set_max_fails(1)` in the console
- Run the command "R: Test R Package"
- Notice that although this package has 4 failing tests, only ones from the first file are run. You'll see output like this:

> Maximum number of failures exceeded; quitting at end of file.
> ℹ Increase this number with (e.g.) `testthat::set_max_fails(Inf)`
